Recursion/FindSqrt.java

```diff
@@ -0,0 +1,35 @@
+import java.math.BigInteger;
+
+public class FindSqrt {
+    
+
+    public static void main(String[] args) {
+        
+
+    }
+
+    public int mySqrt(int x) {
+        BigInteger c = BigInteger.valueOf(x);
+        return c.sqrt().intValue();
+    }
+
+    public int mySqrt2(int x) {
+        if(x == 0){
+            return 0;
+        }
+        int start = 1;
+        int end = x;
+        int ans = 0;
+        while(start <= end){
+            int mid = start + (end-start)/2;
+			//instead of mid*mid we are giving x/mid to tackle overflow of integer range when multiplying with bigger numbers
+            if(mid <= x/mid){
```

Comment:
26  :  potential  infinite  loop  if  x  is  negative  in  my S qrt 2 ()  method